source/api.py

```diff
@@ -292,21 +293,44 @@ def processPendingEvents(processEventQueue=True):
 	if processEventQueue:
 		queueHandler.flushQueue(queueHandler.eventQueue)
 
-def copyToClip(text):
+
+def copyToClip(text, notify=False):
 	"""Copies the given text to the windows clipboard.
 @returns: True if it succeeds, False otherwise.
 @rtype: boolean
 @param text: the text which will be copied to the clipboard
 @type text: string
+@param notify: whether to emit a confirmation message
+@type notify: boolean
 """
 	if not isinstance(text,str) or len(text)==0:
 		return False
 	import gui
-	with winUser.openClipboard(gui.mainFrame.Handle):
-		winUser.emptyClipboard()
-		winUser.setClipboardData(winUser.CF_UNICODETEXT,text)
-	got=getClipData()
-	return got == text
+	try:
+		with winUser.openClipboard(gui.mainFrame.Handle):
+			winUser.emptyClipboard()
+			winUser.setClipboardData(winUser.CF_UNICODETEXT, text)
+		got = getClipData()
+	except ctypes.WinError:
+		if notify:
+			# Translators: Presented when unable to copy to the clipboard
+			# because of an error.
+			ui.message(_("Unable to copy"))
+		return False
+	if got == text:
+		if notify:
+			# Translators: Announced when a text has been copied to clipboard.
+			# %s is replaced by the copied text.
+			speech.speakMessage(_("Copied to clipboard: %s" % text))
+			# Translators: Displayed in braille when a text has been copied to clipboard.
+			# %s is replaced by the copied text.
+			braille.handler.message(_("Copied: %s" % text))

```

Comment:
I'd prefer not import speech for this. Instead, I think we should add an optional "brailleMessage" argument to `ui.message`. That way, the detail for how messaging is done is abstracted away. 
Actually, I think this is only place in `api.py` that uses the `ui` module. We have long had requests for greater customization of UI messages collecting this UI together will probably help (also with consistency of messages). Perhaps the messages should be moved into the `ui` module, EG: `def reportTextCopiedToClipboard(text: Optional[str]) -> None`. When text is `None`, "Unable to copy" is reported. This can be called from each place that wishes to notify, rather than doing it in `api`.

Thinking about why these are different, perhaps both speech and braille should simply use `"Copied: %s"`? It's shorter, and arguably "to clipboard" doesn't tell the user anything.

Reply:
I just rebased onto latest master.
Done adding a `brailleText` parameter to `ui.message`.
Done moving "Unable to copy" within `ui.reportTextCopiedToClipboard`.

Regarding the lengthier spoken message, I felt it was closer to what is currently announced eg. in browse mode.
Let me know if you wish I change this anyway.